classes/Controllers/Rest/ThanksRestV2.php

```diff
@@ -158,116 +157,24 @@ public function CreateThankYou(ServerRequestInterface $request, SecurityContext
-
-		$invalid_params = [];
-
-		foreach ($thanked as $offset => $oclass)
-		{
-			$thanked[$offset] = ['oclass' => (int) ($oclass['oclass'] ?? null), 'id' => (int) ($oclass['id'] ?? null)];
-		}
+		unset($post['id']);
+		$post['author'] = $context->GetUser();
 
 		try
 		{
-			$thankables = $this->api->ThankYous()->CreateThankablesFromOClasses($thanked);
-
-			if (count($thankables) === 0)
-			{
-				$invalid_params[] = ['name' => 'thanked', 'reason' => ($this->lmsg)('thankyou.thankyou.thanked.error.empty')];
-			}
-		} catch (UnsupportedOwnerClassException $exception)
-		{
-			$message = ($this->lmsg)('thankyou.thankable.owner_class.error.not_supported');
-			$first   = true;
-			foreach ($this->api->ThankYous()->GetThankableObjectTypes() as $owner_class)
-			{
-				$message .= $first ? $owner_class : ", " . $owner_class;
-				$first   = false;
-			}
-			$invalid_params[] = ['name' => 'thanked', 'reason' => $message];
-		}
-
-		if ($description === '')
-		{
-			$invalid_params[] = ['name' => 'description', 'reason' => ($this->lmsg)('thankyou.thankyou.description.error.empty')];
-		}
-
-		if ($this->api->Configuration()->IsTagsMandatory() && (!isset($tag_ids) || count($tag_ids) === 0))
-		{
-			$invalid_params[] = ['name' => 'tags', 'reason' => ($this->lmsg)('thankyou.thankyou.core_values.empty')];
-		} elseif (isset($tag_ids))
-		{
-			if (!$this->api->Configuration()->IsTagsEnabled())
-			{
-				$invalid_params[] = ['name' => 'tags', 'reason' => ($this->lmsg)('thankyou.thankyou.tags.error.disabled')];
-			} else
-			{
-				if (!is_array($tag_ids))
-				{
-					$invalid_params[] = ['name' => 'tags', 'reason' => ($this->lmsg)('thankyou.thankyou.tags.error.not_array')];
-				}
-
-				$tag_ids_valid = true;
-				foreach ($tag_ids as $tag_id)
-				{
-					if (!is_int($tag_id))
-					{
-						$tag_ids_valid    = false;
-						$invalid_params[] = ['name' => 'tags', 'reason' => ($this->lmsg)('thankyou.thankyou.tags.error.not_integers')];
-						break;
-					}
-				}
-
-				if ($tag_ids_valid)
-				{
-					$tags = $this->api->Tag()->GetTagsById($tag_ids);
-
-					foreach ($tag_ids as $tag_id)
-					{
-						if (!isset($tags[$tag_id]))
-						{
-							$invalid_params[] = ['name' => 'tags', 'reason' => ($this->lmsg)('thankyou.tag.error.id.not_found', $tag_id)];
-						}
-					}
-				}
-			}
-		}
-
-		if (count($invalid_params) > 0)
+			$thank_you = $this->api->ThankYous()->CreateFromArray($post);
+			$this->api->ThankYous()->Save($context, $thank_you);
+		} catch (ValidationException $validation_exception)
 		{
 			return $this->response->GetJsonPrettyResponse([
 				'type'           => 'https://developer.claromentis.com',
 				'title'          => ($this->lmsg)('thankyou.thankyou.error.create'),
 				'status'         => 400,
-				'invalid-params' => $invalid_params
+				'invalid-params' => $validation_exception->GetErrors()
 			], 400);
-		}
-
-		try
+		} catch (ThankYouForbiddenException | ThankYouNotFoundException | TagNotFoundException $exception)
```

Comment:
`ThankYouForbidden` can still be thrown if a guest user hits this endpoint, and `TagNotFoundException` can definitely be thrown if a tag is supplied that doesn't exist. Neither of these are the developer's fault, so these should instead return a 400 response.

Might just be worth splitting `ThankYous\Api::Save()` into `ThankYous\Api::Create()` and `ThankYous\Api::Update()` to avoid forcing these extra exceptions to be handled.

Reply:
Changes can be found in f094fe0d8021f7fa46e7397b31a34076c1183ca7

Reply:
Looks awesome, though introduced a potential bug (mentioned below)